### PR TITLE
Add start_session

### PR DESCRIPTION
**Pull request by cailinw**

### CODE REVIEW

File: banyan/tests/test_sessions.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from banyan import start_session
+
+# TODO: Check that the required environment variables are set before
+# running tests: api_key, user_id, endpoint
+
```
Comment (calebwin) [inline banyan/tests/test_sessions.py]: Can you at least add some documentation here of how you tested this manually?

File: banyan/banyan/utils.py [hunk 2]
```diff
@@ -121,6 +121,13 @@ def get_python_version():
     return platform.python_version()
 
 
+def convert_iso_time(time):
+    """Convert time from ISO-8601 format (YYYY-MM-DDThh:mm:ss.sTZD) to seconds."""
```
Comment (calebwin) [inline banyan/banyan/utils.py]: nit:
```
"""
Convert time from ISO-8601 format (YYYY-MM-DDThh:mm:ss.sTZD)
to seconds.
"""
```
(put a new line after the """ for readability)

File: banyan/banyan/sessions.py [hunk 3]
```diff
@@ -1,8 +1,194 @@
...
+            FunctionName=executor_lambda_function_name,
+            Runtime=f"python{version}",
+            Role=_create_executor_lambda_iam_role(),
+            Handler="executor.lambda_handler",
+            Code={
+                "S3Bucket": bucket_name,
+                "S3Key": s3_key,
+            },
+            Timeout=900,
+            MemorySize=10240,
```
Comment (calebwin) [inline banyan/banyan/sessions.py]: Is 10240 the maximum memory size? it should be

File: banyan/banyan/sessions.py [hunk 4]
```diff
@@ -1,8 +1,194 @@
...
+    basic_lambda_policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+    iam_role_name = "banyan-executor-lambda-role"
+    try:
+        role = _iam_client.create_role(
+            RoleName=iam_role_name,
+            AssumeRolePolicyDocument=json.dumps(assume_role_policy)
+        )
+        _iam_client.get_waiter("role_exists").wait(RoleName=iam_role_name)
+        _iam_client.attach_role_policy(RoleName=iam_role_name, PolicyArn=basic_lambda_policy_arn)
+        time.sleep(5)
```
Comment (calebwin) [inline banyan/banyan/sessions.py]: Is this 5 second thing an artifact from debugging or was it critical for this to work? Is it critical every time or just the first time a user uses Banyan? and if it takes longer than 5s what will happen?

Doing a while loop to ensure that a resource exists shouldn't be hard. We did this for SQS queues in several places.

File: banyan/banyan/sessions.py [hunk 5]
```diff
@@ -1,8 +1,194 @@
...
+    s3_key = zipfile_name + ".zip"
+    try:
+        _s3_client.create_bucket(
+            Bucket=bucket_name,
+            CreateBucketConfiguration={"LocationConstraint": _s3_client.meta.region_name}
+        )
+        _s3_client.get_waiter("bucket_exists").wait(Bucket=bucket_name)
+    except ClientError as error:
+        if not error.response["Error"]["Code"] in ["BucketAlreadyExists", "BucketAlreadyOwnedByYou"]:
+            raise
```
Comment (calebwin) [inline banyan/banyan/sessions.py]: Does this raise `error`? If so that's good

**Review by calebwin — CHANGES_REQUESTED** [review]

File: banyan/tests/test_sessions.py (hunk 1 above)
Comment (cailinw) [reply banyan/tests/test_sessions.py]: Yes, done.

File: banyan/banyan/sessions.py (hunk 3 above)
Comment (cailinw) [reply banyan/banyan/sessions.py]: Yes, 10,240 MB is 10 GB

File: banyan/banyan/sessions.py (hunk 4 above)
Comment (cailinw) [reply banyan/banyan/sessions.py]: That was intentional; I will add a comment.

I think it usually takes 5-10 seconds to reflect, but a try loop would be better to add. I didn't find an easy way to check this so just added a sleep for now.

File: banyan/banyan/sessions.py (hunk 5 above)
Comment (cailinw) [reply banyan/banyan/sessions.py]: Yes, it will re-raise the original error.

**Pull request merged**